nav2_behavior_tree/plugins/condition/goal_reached_condition.cpp

```diff
@@ -60,6 +60,8 @@ class GoalReachedCondition : public BT::ConditionNode
   {
     node_ = config().blackboard->get<rclcpp::Node::SharedPtr>("node");
     node_->get_parameter_or<double>("goal_reached_tol", goal_reached_tol_, 0.25);
+    node_->get_parameter_or<std::string>("global_frame", global_frame_, "map");
```

Comment:
How are you planning on getting these parameters?

It seems like these would better be input ports for the BT node (see this for example https://github.com/ros-planning/navigation2/blob/71f9a62d0f8dcb0267935d6c6068138b5afad3d5/nav2_behavior_tree/plugins/action/spin_action.cpp#L49) with a default value to map and base_link. Then if you wanted to change, you change the frames in the BT XML file. 

I think this could work as well, but you'd need to figure out what namespace this parameter appears in to add to your config file, which actually sounds like the best situation. Figure out where this parameterization is going to be and then add it to our nav2_params.yaml file.

Reply:
I'm not familiar with nav2_behavior_tree so it's going to take a bit of time.

Reply:
So I'd recommend just looking at this `spin_dist` as a literal example. All you need to do is add a string templated `BT::InputPort` to the provide input ports function. The order of args is variable name, default value, description string. Then in the constructor of the node use `getInput` to get the input of the 2 variables to store. Then you're done!

Reply:
@ymd-stella you can also use this as a direct guide: https://github.com/ros-planning/navigation2/pull/1741 as @naiveHobo did this for the other controller based on your request.